drivers/src/mailbox.rs

```diff
@@ -394,15 +431,60 @@ impl MailboxRecvTxn<'_> {
         Ok(())
     }
 
+    /// Sends `data.len()` bytes to the mailbox FIFO
+    /// Transitions from Execute --> RdyForData
     ///
-    /// Transitions from Execute --> Idle
+    /// # Arguments
     ///
-    pub fn complete(&mut self, success: bool) -> CaliptraResult<()> {
+    /// * `data` - data buffer.
+    ///
+    /// # Returns
+    ///
+    /// Status of Operation
+    ///
+    pub fn copy_response(&mut self, data: &[u8]) -> CaliptraResult<()> {
         if self.state != MailboxOpState::Execute {
             return Err(CaliptraError::DRIVER_MAILBOX_INVALID_STATE);
         }
+
+        self.state = MailboxOpState::RdyForDlen;
+        // Set dlen
+        self.write_dlen(data.len() as u32)?;
+
+        self.state = MailboxOpState::RdyForData;
+        // Copy the data
+        self.enqueue(data)
+    }
+
+    /// Sends `data.len()` bytes to the mailbox FIFO.
+    /// Transitions from Execute --> Idle (releases the lock)
+    ///
+    /// # Arguments
+    ///
+    /// * `data` - data buffer.
+    ///
+    /// # Returns
+    ///
+    /// Status of Operation
+    ///
+    pub fn send_response(&mut self, data: &[u8]) -> CaliptraResult<()> {
```

Comment:
Can this function take `mut self` rather than `&mut self`? The transaction should not be used for anything after the response has been sent.

Reply:
Makes sense, I can change that

Reply:
The issue I'm hitting with this is the drop trait does not take ownership and it needs to call complete. The two options I see are to leave this as-is or to add a private function for complete that doesn't take ownership that Drop() and the public complete function (now a wrapper) can call. 

If we go with the latter, Is there a naming convention for anything like that?

fn complete_private(&mut self, success: bool) // <- naming convention for this one
pub fn complete(mut self, success: bool) {
    self.complete_private()
}

Reply:
Ah, makes sense. I'm good either way.

FWIW, elsewhere in caliptra this is complete_internal().

Reply:
Cool I will add this as "complete_private"

Reply:
I entirely meant to type "complete_internal" here...

Reply:
I think it makes sense to leave this out for now. The use within cold reset flow in ROM seems to only have this as a mutable reference so there's an issue with calling these terminal functions. I think leaving this and not touching the ROM code is probably the safer bet at this point.

Reply:
Ok, I'll followup in a future PR.